Shared/mods/deathmatch/logic/lua/LuaBasic.h

```diff
@@ -228,12 +228,25 @@ namespace lua
         return 1;
     }
 
-    // Tuples can be used to return multiple results
+    // Tuples can be used to build fixed-size tables
+    // e.g. `std::tuple<float, int, bool>` will be pushed as { float, int, bool }
```

Comment:
I'm quite confused?
Now a tuple can be either a `fixed size table` and a `way to push multiple results`?

Reply:
If you use a `std::tuple` as a return value for a scripting function for the arg parser, it'll push multiple values

If you use a `std::tuple` as a parameter to `lua::Push` it'll push a table of values

Reply:
This is also intended to make `lua::Push` behave more consistently for non-argparser usages. For example we could use it to set up globals.

Reply:
Okay, I see. But that's kinda inconsistent, isn't it?
Perhaps we could add a `PushTableValue`, which would make sure if T pushes multiple values onto the stack, it would still become one value (by packing the values into a table, just like I do)

Reply:
Well `lua::Push` would now be consistent (apart from the CLuaArgument edge case above...). 

The arg parser would gain some inconsistency, but we'd otherwise not have a good option to return multiple values.

Reply:
Well, what about disallowing tuples inside tables, and instead use `std::array<std::variant<>>` for fixed size tables?

Reply:
That gets fairly weird quickly, since it obscures the intended return value (which has a static type) behind layers of `variant`. 

We could also add some special multi-return tuple proxy struct, which just acts as a special tuple gadget. That way you could either return a `std::tuple` (and get a table of values) or a `CLuaMultiReturn` (and get multiple values).